app/components/flash-display.js

```diff
@@ -1,15 +1,21 @@
 import Component from '@ember/component';
+import { computed } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { alias } from '@ember/object/computed';
 
 export default Component.extend({
   flashes: service(),
 
-  classNames: ['flash'],
+  classNameBindings: ['type'],
   tagName: 'ul',
 
   messages: alias('flashes.messages'),
 
+  type: computed('messages.@each.type', function () {
+    const classes = this.messages.uniqBy('type').mapBy('type').join(' ');
+    return `flash ${classes}`;
+  }),
```

Comment:
It seems like this change is not relevant anymore, because your custom flash will declare it's styles and appearance separately, in its own template. And all other old types shall work as previously. Could you please revert it (and all related changes)?

Reply:
We still need this class for an additional margin for `warning` Flash (as it's not closable) at the highest level of Flash DOM (outer ul element), but I refactored it and added an optional className for the Flash object.